src/Compilers/Core/Portable/SourceGeneration/GeneratorDriverState.cs

```diff
@@ -95,6 +99,7 @@ namespace Microsoft.CodeAnalysis
             ImmutableArray<GeneratorState>? generatorStates = null,
             ImmutableArray<AdditionalText>? additionalTexts = null,
             DriverStateTable? stateTable = null,
+            SyntaxStore? syntaxStore = null,
```

Comment:
It would be nice if this and 'GeneratorState' were records. I don't think we've had many opportunities to dogfood records.

Reply:
Agreed. I think in the distant past there was even a PROTOTYPE comment about doing that, but the features went in side by side and we didn't get a chance to switch over. I've opened https://github.com/dotnet/roslyn/issues/59199 to track doing that, but I'll keep it separate from this PR to reduce churn in an already churn-y PR.